packages/ecr-credential-provider/ecr-credential-provider.spec

```diff
@@ -0,0 +1,45 @@
+%global goproject github.com/kubernetes
+%global gorepo cloud-provider-aws
+%global goimport %{goproject}/%{gorepo}
+
+%global gover 1.24.1
+%global rpmver %{gover}
+
+%global gitrev b9c8d82db9edc3be45f82f8d5320d006c62aa50c
+%global shortrev %(c=%{gitrev}; echo ${c:0:7})
+
+%global _missing_build_ids_terminate_build 0
+%define debug_package %{nil}
+
+Name: %{_cross_os}aws-ecr-credential-provider
+Version: %{rpmver}
+Release: 1%{?dist}
+Summary: Container image registry credential provider for AWS ECR
+License: Apache-2.0
+URL: https://github.com/kubernetes/cloud-provider-aws
+Source: bundled-v%{gover}.tar.gz
+Source1: v%{gover}.tar.gz
+Source1000: clarify.toml
+
+%description
+%{summary}.
+
+%prep
+%setup -n %{gorepo}-%{gitrev} -q
+%setup -T -D -n %{gorepo}-%{gitrev} -b 1 -q
+
+%build
+%set_cross_go_flags
+make ecr-credential-provider GIT_VERSION=%{gover}
```

Comment:
I read the `Makefile` and it seems like `ecr-credential-provider` is statically linked since in the `Makefile` `CGO_ENABLED` is set to `0`. Another flag set in the `Makefile` is `-ldflags="$(LDFLAGS)"` and that doesn't match what is used in `hotdog` (the other statically linked go package included in Bottlerocket), which is just `-ldflags "-s"`. Even if we don't want to set that value for `-ldflags`, it will be best to use what other go packages use (`-ldflags "${GOLDFLAGS}"`). Thus, I think it will be better if we don't use the `Makefile` and instead we craft the compilation command to set the flags to the values we need.

Reply:
I understand it is a little odd to not honor the `Makefile` provided by the upstream sources, but there are already packages in Bottlerocket where we change the upstream behavior to add additional flags to improve the security of the binaries, since we might have a different opinion on what a good set of default flags is for a given binary. Example packages where we don't use `Makefile`s provided by the upstream sources include [docker-cli], [docker-engine], [containerd], [nvidia-container-toolkit]. We even go a step further in the [libnvidia-container] spec, and we patch the `Makefile` to do what works best for Bottlerocket.

[docker-cli]: https://github.com/bottlerocket-os/bottlerocket/blob/develop/packages/docker-cli/docker-cli.spec
[docker-engine]: https://github.com/bottlerocket-os/bottlerocket/blob/develop/packages/docker-engine/docker-engine.spec
[containerd]: https://github.com/bottlerocket-os/bottlerocket/blob/develop/packages/containerd/containerd.spec
[nvidia-container-toolkit]: https://github.com/bottlerocket-os/bottlerocket/blob/develop/packages/nvidia-container-toolkit/nvidia-container-toolkit.spec
[libnvidia-container]: https://github.com/bottlerocket-os/bottlerocket/blob/develop/packages/libnvidia-container/libnvidia-container.spec

In the `Makefile` of `ecr-credential-provider`, they are just overriding `ldflags` with whatever `LDFLAGS` has, which in Bottlerocket's build system might be empty or set to the wrong flags, since the packages should be built with `"%{_cross_ldflags}"` so that cross compilation works as expected.

> We don't want to get in a situation where ecr-credentail-provider in bottlerocket has a subtly different behavior than what someone would get from pulling down the binary from the upstream project.

I understand this concern, and there are packages in Bottlerocket where we couldn't use the flags we wanted. One example of this is [nvidia-k8s-device-plugin], where we couldn't add the hardening flags since otherwise the libraries the binary depends on weren't loaded. But I caught that problem through testing, to make sure the package works as expected.

[nvidia-k8s-device-plugin]: https://github.com/bottlerocket-os/bottlerocket/blob/develop/packages/nvidia-k8s-device-plugin/nvidia-k8s-device-plugin.spec#L28

Reply:
OK, that seems reasonable. I'll update this to follow what we're doing elsewhere then.